### ISSUE TITLE
Deleting/moving rotated files while the process is running leads to panic

### ISSUE THREAD

**Issue by mbiggio** [BourgondAries/file-rotate#17]
I noticed that `FileRotate` panics if the rotated files are deleted or moved away from the log folder while the application logging through `FileRotate` is still running.

More precisely, the crash is caused by [this assert](https://github.com/BourgondAries/file-rotate/blob/master/src/lib.rs#L531), in which the code is checking that the file to be renamed in a rotation still exists.

I think the reason for that assert is that `FileRotate` makes a snapshot of the currently rotated files in the log folder [when the `scan_suffixes` method is called in the constructor](https://github.com/BourgondAries/file-rotate/blob/master/src/lib.rs#L426), and then assumes that the rotated files will not be moved/deleted by an external process.

Do you think it might be possible to lighen this assumption and be more robust to this usecase?
Thanks for your input on this :slightly_smiling_face:

**Comment by Ploppz** [comment]
Yeah, that should be made more robust. I guess instead of the assert, we should just call `scan_suffixes` again if those assumptions are not true.
I'll get to this next week probably, unless there be a PR before that time ;)

**Comment by Ploppz** [comment]
I started work on this feature

**Comment by Ploppz** [comment]
Current blocker is concern about race conditions in my PR.
I wonder if we should use lock files or something, or if we can solve it in another way.
